### DISCUSSION TITLE
Error on NextJS build

### DISCUSSION THREAD

**Discussion by ismi-abbas** [shadcn-ui/ui#2337] [General]
Using 
```
export const fontSans = FontSans({
  subsets: ['latin'],
  variable: '--font-sans',
})
```
on the root layout will throw type error during build

```
Type error: Type 'OmitWithTag<typeof import("C:/Users/test/projects/nextjs-event-manager/app/layout"), "metadata" | "default" | "config" | "generateStaticParams" | "revalidate" | "dynamic" | "dynamicParams" | ... 6 more ... | "generateViewport", "">' does not satisfy the constraint '{ [x: string]: never; }'.
  Property 'fontSans' is incompatible with index signature.
    Type 'NextFontWithVariable' is not assignable to type 'never'.

   6 |
   7 | // Check that the entry is a valid entry
>  8 | checkFields<Diff<{
     |             ^
   9 |   default: Function
  10 |   config?: {}
  11 |   generateStaticParams?: Function
```
Removing export will make it success

Temporary fix
```
const fontSans = FontSans({
  subsets: ['latin'],
  variable: '--font-sans',
})
```
Refer [NextFont gives a type error on Build mode](https://github.com/vercel/next.js/discussions/58693)

**Comment by shadcn** [comment]
@ismi-abbas Where are you placing the font export? Which file?

**Comment by ismi-abbas** [comment]
> @ismi-abbas Where are you placing the font export? Which file?

At the app/layout.tsx. I've initiated the ui with src folder, then I remove it, configured all the path to be correct, but still encounter problem.

```
{
  "$schema": "https://ui.shadcn.com/schema.json",
  "style": "new-york",
  "rsc": true,
  "tsx": true,
  "tailwind": {
    "config": "tailwind.config.ts",
    "css": "app/globals.css",
    "baseColor": "zinc",
    "cssVariables": true,
    "prefix": ""
  },
  "aliases": {
    "components": "@/components",
    "utils": "@/lib/utils"
  }
}
```

**Comment by shadcn** [comment]
Yeah you can't export from the `layout.tsx` file. I usually keep a `lib/fonts.ts` file where I keep all font exports.

**Reply by emre-turan** [reply]
I found myself in the same situation, but I believe we should be more cautious when copying and pasting. However, there is room for improvement in the instructions provided here.

<img width="726" alt="Screenshot 2024-01-09 at 11 32 39" src="https://github.com/shadcn-ui/ui/assets/82906617/a13fb735-5a85-4435-b540-452ff501b441">
